Neos.Fusion/Classes/Core/Runtime.php

```diff
@@ -399,25 +440,25 @@ protected function evaluateInternal($fusionPath, $behaviorIfPathNotFound, $conte
             if ($this->hasExpressionOrValue($fusionConfiguration)) {
                 return $this->evaluteExpressionOrValueInternal($fusionPath, $fusionConfiguration, $cacheContext, $contextObject);
```

Comment:
Should spreads also somehow be involved here? I seem to miss how a spread would affect these values if not given to the fusion object but evaluated "lazily". "setPropertiesOnFusionObject" is only called for array type fusion objects. Anything else will just internally evaluate fusion paths via runtime and probably miss the spreads? Might need to see if I can construct a test case. I am also not sure if there isn't an easier way than a separate spread stack, but that might actually be the only way...

Reply:
The spreads are checked earlier in evaluateInternal and thus this code is not reached if a spread value for this key was present.

Reply:
The setFusionProperties was actually implemented at last and only notifies the array fusion objects about available keys. Even the array objects fetch the actual values via evaluate.